### DISCUSSION TITLE
Need help integrating AzuraCast with Keycloak

### DISCUSSION THREAD

**Discussion by alkahtani14** [AzuraCast/AzuraCast#6435] [Q&A]
Hello,

I'm reaching out to the AzuraCast community for assistance with integrating AzuraCast with Keycloak for my online radio project. First and foremost, I want to express my immense gratitude to the AzuraCast team and contributors for their incredible work on this open-source project. AzuraCast has been instrumental in enabling me to launch and manage my online radio station.

Now, I'm facing some challenges in integrating Keycloak with AzuraCast, and I would be extremely grateful for any help or guidance from the community.

Specifically, I need assistance with the following:

1. Configuring AzuraCast to work with Keycloak as the authentication provider.
2. Implementing single sign-on (SSO) between AzuraCast and Keycloak.
3. Managing user roles and permissions through Keycloak.

I have explored the available documentation and resources, but I couldn't find a comprehensive guide for this integration. If anyone has prior experience or knowledge in integrating AzuraCast with Keycloak, I would be deeply thankful for your insights and assistance.

Here are some additional details that might be useful:

- AzuraCast version: 0.18.5
- Keycloak version: 22.0.0

Once again, I want to express my heartfelt gratitude to the AzuraCast team and community for their dedication to this project. Thank you for considering my request, and I look forward to any guidance or support you can provide.

**Comment by BusterNeece** [comment]
Thank you for reaching out! To answer your question, right now AzuraCast has no direct way to tie in to Keycloak as its authentication provider. I've looked into Keycloak, and while it certainly seems like powerful and useful software, it's not clear to me how to integrate it into our application simply.

I'd recommend submitting a new feature request for this implementation over at https://features.azuracast.com/, but in the interest of full transparency, currently this isn't a high priority for me to implement.
